Spark History Server: add compatibility with S3

With current version of Spark History Server (docker image: saagie/spark-history-server:1.1) , we can't use AWS S3 as a location of event log of Spark. 

`Exception in thread "main" java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.apache.spark.deploy.history.HistoryServer$.main(HistoryServer.scala:296)
	at org.apache.spark.deploy.history.HistoryServer.main(HistoryServer.scala)
Caused by: java.lang.RuntimeException: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
	at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2154)
	at org.apache.hadoop.fs.FileSystem.getFileSystemClass(FileSystem.java:2580)
	at org.apache.hadoop.fs.FileSystem.createFileSystem(FileSystem.java:2593)
	at org.apache.hadoop.fs.FileSystem.access$200(FileSystem.java:91)
	at org.apache.hadoop.fs.FileSystem$Cache.getInternal(FileSystem.java:2632)
	at org.apache.hadoop.fs.FileSystem$Cache.get(FileSystem.java:2614)
	at org.apache.hadoop.fs.FileSystem.get(FileSystem.java:370)
	at org.apache.hadoop.fs.Path.getFileSystem(Path.java:296)
	at org.apache.spark.deploy.history.FsHistoryProvider.<init>(FsHistoryProvider.scala:117)
	at org.apache.spark.deploy.history.FsHistoryProvider.<init>(FsHistoryProvider.scala:86)
	... 6 more
Caused by: java.lang.ClassNotFoundException: Class org.apache.hadoop.fs.s3a.S3AFileSystem not found
	at org.apache.hadoop.conf.Configuration.getClassByName(Configuration.java:2060)
	at org.apache.hadoop.conf.Configuration.getClass(Configuration.java:2152)
	... 15 more `

--- a/technologies/app/spark-history-server/README.md
+++ b/technologies/app/spark-history-server/README.md
@@ -1,6 +1,6 @@
 # Spark History Server
 
-![Docker Image Size (tag)](https://img.shields.io/docker/image-size/saagie/spark-history-server/1.1?label=v1.1%20image%20size&style=for-the-badge)
+![Docker Image Size (tag)](https://img.shields.io/docker/image-size/saagie/spark-history-server/1.2?label=v1.2%20image%20size&style=for-the-badge)
 
 ## Description
 This directory contains version of Spark History Server contenairized and customized for Saagie Platform.

--- a/technologies/app/spark-history-server/metadata.yaml
+++ b/technologies/app/spark-history-server/metadata.yaml
@@ -21,4 +21,4 @@ contexts:
         basePath: SAAGIE_BASE_PATH
     dockerInfo:
       image: saagie/spark-history-server
-      version: 1.1
+      version: 1.2

--- a/technologies/app/spark-history-server/spark-history-server/Dockerfile
+++ b/technologies/app/spark-history-server/spark-history-server/Dockerfile
@@ -1,4 +1,4 @@
-FROM saagie/spark:3.0-1.69.0 as builder
+FROM saagie/spark:3.0-aws-0.11.2 as builder
 
 FROM alpine:3.10
 

--- a/version.properties
+++ b/version.properties
@@ -1,6 +1,6 @@
-version.buildmeta=
+version.buildmeta=bug/59-spark-history-add-compatibility-s3
 version.major=0
 version.minor=17
 version.patch=0
-version.prerelease=
-version.semver=0.17.0
+version.prerelease=BETA
+version.semver=0.17.0-BETA+bug/59-spark-history-add-compatibility-s3

